test/workspaces/coverage-report-tests/check-coverage.test.ts

```diff
@@ -0,0 +1,46 @@
+import { existsSync, readFileSync } from 'node:fs'
+import { normalize } from 'node:path'
+import { expect, test } from 'vitest'
+import libCoverage from 'istanbul-lib-coverage'
+import { resolve } from 'pathe'
+
+test('coverage exists', () => {
+  expect(existsSync('./coverage')).toBe(true)
+  expect(existsSync('./coverage/index.html')).toBe(true)
+})
+
+test('file coverage summary matches', () => {
+  const coverageJson = JSON.parse(readFileSync('./coverage/coverage-final.json', 'utf-8'))
+  const coverageMap = libCoverage.createCoverageMap(coverageJson)
+  const fileCoverage = coverageMap.fileCoverageFor(normalize(resolve('./src/math.ts')))
+
+  // There should be 1 uncovered branch and 1 uncovered function. See math.ts.
+  expect(fileCoverage.toSummary()).toMatchInlineSnapshot(`
+    {
+      "branches": {
+        "covered": 3,
+        "pct": 75,
+        "skipped": 0,
+        "total": 4,
+      },
+      "functions": {
+        "covered": 2,
+        "pct": 66.66,
+        "skipped": 0,
+        "total": 3,
```

Comment:
It looks like the function count does not match. Weird that no-one has reported this issue. 
We'll need to add `/* c8 ignore next */` above each `Object.defineProperty(__vite_ssr_exports__, ...`. I'll work on that on separate PR as it's unrelated to this issue.

Reply:
Fixed in https://github.com/vitest-dev/vitest/pull/3241